scipy/special/xsf/binom.h

```diff
@@ -86,4 +88,40 @@ XSF_HOST_DEVICE inline float binom(float n, float k) {
     return binom(static_cast<double>(n), static_cast<double>(k));
 }
 
+XSF_HOST_DEVICE inline double binomln(double n, double k) {
+    /* Natural logarithm of absolute value of binomial coefficient */
+    double a = n - k + 1;
+    bool denominator_pole = (k <= - 1 && k == std::floor(k)) || (a <= 0 && a == std::floor(a));
+    if (n <= -1 && n == std::floor(n)) {
+	return denominator_pole ? std::numeric_limits<double>::quiet_NaN() : std::numeric_limits<double>::infinity();
+    }
+    if (denominator_pole) {
+	return -std::numeric_limits<double>::infinity();
+    }
+    if (n == k) {
+	return 0;
+    }
```

Comment:
Is there similar logic in `binom`? That is, can we handle these cases with with `log(binom(n, k))`?

Reply:
K, the thing is that I need to review it. What should I use as a reference?
Or would the expression in terms of `betaln` do the right thing?